app/controllers/government_rulesets_controller.rb

```diff
@@ -4,17 +4,39 @@ class GovernmentRulesetsController < ApplicationController
   before_action :clear_errors
 
   def show
-    upload = Upload.find(session[:upload_id])
-    return redirect_to root_path, alert: "Please re upload your file" if upload.nil?
+    @upload = Upload.find(session[:upload_id])
+    return redirect_to root_path, alert: "Please re-upload your file" if upload.nil?
 
-    @spectral_results =
-      JSON.parse(
-        Linters::Spectral.new(upload:).lint_to_json
-      )
+    @issues = []
+    @score = 100
+    @counter = 0
+    spectral_hash = JSON.parse(spectral_output)
+    spectral_hash.each do |issue|
+      newissue = {
+        code: issue["code"],
+        path: issue["path"],
+        message: issue["message"],
+        criticality: 5 - issue["severity"],
+        range: issue["range"]
+      }
+      @counter += 1
```

Comment:
[suggestion]
Wouldn't it be simpler to just use `@issues.length` (or `count`, or `size`) rather than maintaining a separate counter flag

Reply:
Fixed. I am not sure why I had this - I think I was doing something differently earlier with the sorting, but I agree it is pointless now.